projects/kit/components/rating/rating.component.ts

```diff
@@ -0,0 +1,152 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    EventEmitter,
+    forwardRef,
+    HostListener,
+    Inject,
+    Input,
+    Output,
+} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+import {EMPTY_FUNCTION, moveFocus} from '@taiga-ui/cdk';
+
+import {RatingOptions, TUI_RATING_OPTIONS} from './rating.options';
+import {TuiRatingState} from './rating.state';
+import {getChildrenByParent} from './utils/get-children-by-parent';
+
+@Component({
+    selector: 'tui-rating',
+    templateUrl: './rating.template.html',
+    styleUrls: ['./rating.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => TuiRatingComponent),
+            multi: true,
+        },
+    ],
+})
+export class TuiRatingComponent implements ControlValueAccessor {
+    @Input() tuiRateMax: number = this.options.rateMax;
```

Comment:
Prefixes only make sense for directives to not clash with other inputs, in components we do not need them because there can be only one component per element.